### ISSUE TITLE
Double softmax

### ISSUE THREAD

**Issue by ceteke** [fchollet/keras#6897]
Hi,

I was looking at Nietzsche text generation example (https://github.com/fchollet/keras/blob/master/examples/lstm_text_generation.py). I have seen that after predicting, `sample(preds, temperature=1.0)` function is used which is a softmax. But the network has softmax as an activation function. So after prediction a redundant softmax is applied. Is this on purpose or is it a mistake? If it is not a mistake can you explain the reasoning behind it?

Thanks

**Comment by the-moliver** [comment]
It's not a mistake and is correct as is. Notice the log transformation before the second softmax https://github.com/fchollet/keras/blob/master/examples/lstm_text_generation.py#L65, which undoes the exponentiation from the original softmax. This is then divided by the temperature and fed through the second softmax to adjust the relative probability of each output. See:
https://en.wikipedia.org/wiki/Softmax_function#Reinforcement_learning

Please close this issue.
